### PR TITLE
#830 - Add home page

### LINKED ISSUE #830: Control home page
We need to create a new control home page that serves as a step where you choose whether you want to view recipes or extensions.

### PR DESCRIPTION
**Pull request by andymikulski**
Fixes #830 ~, requires PR #837 to land.~

- Adds new home page to index route
  - Presents user with buttons to navigate to the Recipes or Extensions listings
- Adds decorator support
  - Adds `@autobind`
- Adds `babel-plugin-import` to suppress ant console warnings
- Adds `prop-types` package
  - Replaces deprecated PropTypes with new hotness

~**NOTE** This probably will require a rebase onto Rehan's router PR (#837)~

### CODE REVIEW

**Comment by rehandalal** [comment]
It does in fact need a rebase to match the new routing pattern.

File: recipe-server/client/control_new/components/Gateway.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+  render() {
+    return (
+      <Row className="page-gateway" type="flex" justify="space-around" align="top">
+        <Col className="gw-col" span={8} onClick={this.gotoRecipes}>
+          <h2 className="title">Recipes</h2>
+          <Icon type="book" />
+          <p>This is for normal recipes and stuff or whatever.</p>
+        </Col>
+        <Col className="gw-col" span={8} onClick={this.gotoExtensions}>
```
Comment (mythmon) [inline recipe-server/client/control_new/components/Gateway.js]: We should probably be using `<Link>` elements from redux-little-router. They properly handle opening things in new tabs for middle-clicks and control-clicks.

Comment (andymikulski) [inline recipe-server/client/control_new/components/Gateway.js]: I changed this _moments_ after that discussion was had, but the way Ant handles `Link`s breaks the formatting, so I didn't push it. I'll correct after the rebase and get that up.

Comment (mythmon) [inline recipe-server/client/control_new/components/Gateway.js]: Alternatively, we could also handle the onlick stuff ourself.

File: recipe-server/client/control_new/less/pages/gateway.less [hunk 2]
```diff
@@ -0,0 +1,31 @@
+.page-gateway {
+  .anticon {
+    display: block;
+    font-size: 3em;
+    margin-bottom: 0.5em;
+    text-align: center;
+  }
+
+  .gw-col {
+    background: #E1F4CB;
```
Comment (andymikulski) [inline recipe-server/client/control_new/less/pages/gateway.less]: Based on [this blog post](https://medium.com/@GeoffMiller/how-to-customize-ant-design-with-react-webpack-the-missing-guide-c6430f2db10f), I think it'd be better to make color variables its own PR. Looks like we have to play with Webpack to get it to play nice with Less.

File: recipe-server/client/control_new/components/Gateway.js (hunk 1 above)
Comment (andymikulski) [inline recipe-server/client/control_new/components/Gateway.js]: I've switched to using a `Link` inside the component. This lets us ctrl+click, as well as use `tab` to change link focus.

File: recipe-server/client/control_new/less/pages/gateway.less (hunk 2 above)
Comment (andymikulski) [inline recipe-server/client/control_new/less/pages/gateway.less]: After discussion and some testing, we actually don't need to do any of the Webpack business. I've converted this over to rely on Ant's color variables.

File: recipe-server/package.json [hunk 3]
```diff
@@ -17,6 +17,8 @@
   "license": "MPL-2.0",
   "dependencies": {
     "antd": "2.11.2",
+    "autobind-decorator": "^1.4.4",
```
Comment (mythmon) [inline recipe-server/package.json]: We should pin versions.

File: recipe-server/client/control_new/components/App.js [hunk 4]
```diff
@@ -1,11 +1,10 @@
 import { Breadcrumb, Layout, LocaleProvider } from 'antd';
 import enUS from 'antd/lib/locale-provider/en_US';
-import React, { PropTypes as pt } from 'react';
-
+import React from 'react';
+import PropTypes from 'prop-types';
```
Comment (rehandalal) [inline recipe-server/client/control_new/components/App.js]: 👍🏼

File: recipe-server/client/control_new/components/data/QueryExtensions.js [hunk 5]
```diff
@@ -1,4 +1,5 @@
-import React, { PropTypes as pt } from 'react';
+import React from 'react';
+import PropTypes from 'prop-types';
```
Comment (rehandalal) [inline recipe-server/client/control_new/components/data/QueryExtensions.js]: ordering

**Comment by andymikulski** [comment]
@rehandalal I've swapped out the elements on the Gateway with Ant's `Card`s - 

screen shot 2017-06-28 at 2 24 16 pm


I've also swapped out the broken Ant icon.
